Microsoft.Toolkit.Uwp.UI/Media/CanvasBrushBase.cs

```diff
@@ -38,12 +38,24 @@ public abstract class CanvasBrushBase : XamlCompositionBrushBase
         /// <returns>True if drawing was completed and the brush is ready, otherwise return False to not create brush yet.</returns>
         protected abstract bool OnDraw(CanvasDevice device, CanvasDrawingSession session, Vector2 size);
 
+        /// <summary>
+        /// Gets or sets the shared device.
+        /// </summary>
+        protected CanvasDevice Device { get; set; }
```

Comment:
You shouldn't have to expose these, all the event handling should just happen in this base class, the RadialGradientBrush shouldn't have to know about any of these things.